src/app/screens/Register/index.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import i18next from 'i18next';
+import { useForm } from 'react-hook-form';
+
+import AuthWrapper from '~components/AuthWrapper';
+import Input from '~components/Input';
+import { emailRegex } from '~utils/inputValidations';
+import { AUTH_FIELDS } from '~constants/fields';
+
+import styles from './styles.module.scss';
+
+interface DataForm {
+  email: string;
+  firstName: string;
+  lastName: string;
+  locale: string;
+  password: string;
+  passwordConfirmation: string;
+}
+
+function Register() {
+  const { register, handleSubmit, errors, formState, getValues } = useForm({
+    mode: 'all'
+  });
+
+  const submitForm = (data: DataForm) => {
```

Comment:
the type for react-hook-form goes in the definition

```
const { register, handleSubmit, errors, formState, getValues } = useForm<DataForm>({
    mode: 'all'
  });
```
So you don't have to type the submit args because its inferred. Also the errors